src/hera/workflows/artifact.py

```diff
@@ -73,7 +74,7 @@ class Artifact(BaseModel):
     path: Optional[str] = None
     """path where the artifact should be placed/loaded from"""
 
-    recurse_mode: Optional[str] = None
```

Comment:
@flaviuvadan what would be "both"? This type was always incorrectly set

Reply:
Considering we've had this code the whole time:
```py
    def _build_artifact(self) -> _ModelArtifact:
        ...
        return _ModelArtifact(
            ...
            recurse_mode=self.recurse_mode,
```
Which requires the `recurse_mode` to be the literal string of `True`/`true` etc, it just means we're bringing forward the point of failure for the user, i.e. there will be no behaviour change here. Tested:

```py
>>> from hera.shared._pydantic import BaseModel
>>> class MyClass(BaseModel):
...     my_bool_as_str: str  # This will be converted to a bool later
...
>>> MyClass(my_bool_as_str="true")
MyClass(my_bool_as_str='true')
>>> MyClass(my_bool_as_str="True")
MyClass(my_bool_as_str='True')
>>> class MyClass(BaseModel):
...     my_bool_as_str: bool  # this is now converting to bool immediately
...
>>> MyClass(my_bool_as_str="true")
MyClass(my_bool_as_str=True)
>>> MyClass(my_bool_as_str="True")
MyClass(my_bool_as_str=True)
```

Reply:
Was thinking bool and str